nupic/research/monitor_mixin/temporal_memory_monitor_mixin.py

```diff
@@ -209,10 +215,8 @@ def mmPrettyPrintConnections(self):
           synapseList = []
 
           for synapse in self.connections.synapsesForSegment(seg):
-            (_, sourceCell, permanence) = self.connections.dataForSynapse(
-              synapse)
-
-            synapseList.append((sourceCell, permanence))
+            synapseData = self.connections.dataForSynapse(synapse)
+            synapseList.append((synapseData.presynapticCell, synapseData.permanence))
```

Comment:
With the C++ TM, this prefers synapseData.presynapticCell.idx (or it prints bad index during connections pretty printing).